File: plot/src/main/java/net/fwitz/math/plot/binary/dual/color/DualNumberColorFunction.java
package net.fwitz.math.plot.binary.dual.color;

import net.fwitz.math.binary.dual.DualNumber;

import java.awt.*;
import java.util.function.BiFunction;

@FunctionalInterface
public interface DualNumberColorFunction extends BiFunction<DualNumber, DualNumber, Color> {
    /**
     * Calculate the color for the given input and output values.
     *
     * @param c the input value
     * @param z the output value
     * @return the color to paint
     */
    @Override
    Color apply(DualNumber c, DualNumber z);
}

File: plot/src/main/kotlin/net/fwitz/math/plot/binary/dual/color/DualNumberColorFunction.kt
package net.fwitz.math.plot.binary.dual.color

import net.fwitz.math.binary.dual.DualNumber
import net.fwitz.math.plot.binary.dual.DualNumberFunctionPlot
import java.awt.Color
import java.util.function.BiFunction

@FunctionalInterface
interface DualNumberColorFunction : (DualNumber, DualNumber) -> Color {
    /**
     * Calculate the color for the given input and output values.
     *
     * @param c the input value
     * @param z the output value
     * @return the color to paint
     */
    override fun invoke(c: DualNumber, z: DualNumber): Color
}